`make unittest` fails after swift-DEVELOPMENT-SNAPSHOT-2025-05-23-a

```
Packaging tests...
[1/21] .build/plugins/PackageToJS/outputs/PackageTests/package.json: skipped
[2/21] .build/plugins/PackageToJS/outputs/PackageTests.tmp/npm-install.stamp: skipped
[3/21] .build/plugins/PackageToJS/outputs/PackageTests/JavaScriptKitPackageTests.xctest.wasm: building
[4/21] .build/plugins/PackageToJS/outputs/PackageTests.tmp/wasm-imports.json: building
[5/21] .build/plugins/PackageToJS/outputs/PackageTests/bin: skipped
[6/21] .build/plugins/PackageToJS/outputs/PackageTests/bin/test.js: skipped
[7/21] .build/plugins/PackageToJS/outputs/PackageTests/bridge.js: skipped
[8/21] .build/plugins/PackageToJS/outputs/PackageTests/index.d.ts: building
[9/21] .build/plugins/PackageToJS/outputs/PackageTests/index.js: building
[10/21] .build/plugins/PackageToJS/outputs/PackageTests/instantiate.d.ts: building
[11/21] .build/plugins/PackageToJS/outputs/PackageTests/instantiate.js: building
[12/21] .build/plugins/PackageToJS/outputs/PackageTests/platforms/browser.d.ts: building
[13/21] .build/plugins/PackageToJS/outputs/PackageTests/platforms/browser.js: building
[14/21] .build/plugins/PackageToJS/outputs/PackageTests/platforms/browser.worker.js: building
[15/21] .build/plugins/PackageToJS/outputs/PackageTests/platforms/node.d.ts: building
[16/21] .build/plugins/PackageToJS/outputs/PackageTests/platforms/node.js: building
[17/21] .build/plugins/PackageToJS/outputs/PackageTests/runtime.d.ts: building
[18/21] .build/plugins/PackageToJS/outputs/PackageTests/runtime.js: building
[19/21] .build/plugins/PackageToJS/outputs/PackageTests/test.browser.html: skipped
[20/21] .build/plugins/PackageToJS/outputs/PackageTests/test.d.ts: skipped
[21/21] .build/plugins/PackageToJS/outputs/PackageTests/test.js: skipped
Packaging tests finished
Running test...
$ "/home/katei/.config/nvm/versions/node/v22.9.0/bin/node" "--experimental-wasi-unstable-preview1" "/home/katei/ghq/github.com/swiftwasm/JavaScriptKit/.build/plugins/PackageToJS/outputs/PackageTests/bin/test.js" "--prelude" "/home/katei/ghq/github.com/swiftwasm/JavaScriptKit/Tests/prelude.mjs" "--"

Test Suites: 1 unknown, 1 total
Tests:       1 unknown, 1 total
Ran all test suites.

Failed test cases:
  ? JSObjectTests.testEmptyObject

Some tests failed. Use --verbose for raw test output.
Running test...
$ "/home/katei/.config/nvm/versions/node/v22.9.0/bin/node" "--experimental-wasi-unstable-preview1" "/home/katei/ghq/github.com/swiftwasm/JavaScriptKit/.build/plugins/PackageToJS/outputs/PackageTests/bin/test.js" "--prelude" "/home/katei/ghq/github.com/swiftwasm/JavaScriptKit/Tests/prelude.mjs" "--" "--testing-library" "swift-testing"
◇ Test run started.
↳ Testing Library Version: 6.3-dev (981aa1c64215e71)
↳ Target Platform: wasm32-unknown-wasi
✔ Test run with 0 tests in 0 suites passed after 0.006918866 seconds.
wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:1


RuntimeError: memory access out of bounds
    at JavaScriptKitPackageTests.xctest._swift_release_dealloc (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[28629]:0x529ef3)
    at JavaScriptKitPackageTests.xctest.swift_nonatomic_release (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[28616]:0x5296aa)
    at JavaScriptKitPackageTests.xctest.swift_release (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[28619]:0x529984)
    at JavaScriptKitPackageTests.xctest..Lobjectdestroy (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[87309]:0x10ee9fb)
    at JavaScriptKitPackageTests.xctest._swift_release_dealloc (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[28629]:0x529ef6)
    at JavaScriptKitPackageTests.xctest.swift_nonatomic_release (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[28616]:0x5296aa)
    at JavaScriptKitPackageTests.xctest.swift_release (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[28619]:0x529984)
    at JavaScriptKitPackageTests.xctest.$s7Testing6RunnerVwxx (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[87289]:0x10ecb17)
    at JavaScriptKitPackageTests.xctest.$s7Testing6RunnerVWOh (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[84390]:0x103955a)
    at JavaScriptKitPackageTests.xctest.$s7Testing6RunnerV4_run33_E393D8EDD5AFFA0BF3AEFD347B81DFDFLLyyACYaFZTY4_ (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[87203]:0x10dee81)
    at JavaScriptKitPackageTests.xctest.swift::runJobInEstablishedExecutorContext(swift::Job*) (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[33114]:0x61870a)
    at JavaScriptKitPackageTests.xctest.swift_job_run (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[33133]:0x61946d)
    at JavaScriptKitPackageTests.xctest.$sScJ16runSynchronously2onySce_tF (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[323]:0x575f2)
    at JavaScriptKitPackageTests.xctest.$s19JavaScriptEventLoopAAC10runAllJobsyyF (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[419]:0x5acc3)
    at JavaScriptKitPackageTests.xctest.$s19JavaScriptEventLoopAAC14insertJobQueue3jobyScJ_tFyycfU0_ (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[418]:0x5aab6)
    at JavaScriptKitPackageTests.xctest.$s19JavaScriptEventLoopAAC14insertJobQueue3jobyScJ_tFyycfU0_TA (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[410]:0x5a66a)
    at JavaScriptKitPackageTests.xctest.$s19JavaScriptEventLoopAAC6create33_F9DB15AFB1FFBEDBFE9D13500E01F3F2LLAByFZyyyccfU0_0aB3Kit7JSValueOAGcfU_ (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[308]:0x56be6)
    at JavaScriptKitPackageTests.xctest.$s19JavaScriptEventLoopAAC6create33_F9DB15AFB1FFBEDBFE9D13500E01F3F2LLAByFZyyyccfU0_0aB3Kit7JSValueOAGcfU_TA (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[307]:0x56b23)
    at JavaScriptKitPackageTests.xctest.$s13JavaScriptKit9JSPromiseC4then7successAcA7JSValueOAGc_tFAGSayAGGYucfU_ (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1902]:0xb2a00)
    at JavaScriptKitPackageTests.xctest.$s13JavaScriptKit9JSPromiseC4then7successAcA7JSValueOAGc_tFAGSayAGGYucfU_TA (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1900]:0xb2891)
    at JavaScriptKitPackageTests.xctest.$sSay13JavaScriptKit7JSValueOGACIeggTo_AdCIegxTo_TR (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[314]:0x5719c)
    at JavaScriptKitPackageTests.xctest.$sSay13JavaScriptKit7JSValueOGACIeggTo_AdCIegxTo_TRTA.10 (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1901]:0xb290f)
    at JavaScriptKitPackageTests.xctest.$s13JavaScriptKit16JSOneshotClosureC_4file4lineAcA7JSValueOSayAGGnYuc_SSs6UInt32VtcfcAgHYucfU0_ (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1452]:0x91947)
    at JavaScriptKitPackageTests.xctest.$s13JavaScriptKit16JSOneshotClosureC_4file4lineAcA7JSValueOSayAGGnYuc_SSs6UInt32VtcfcAgHYucfU0_TA (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1448]:0x916c0)
    at JavaScriptKitPackageTests.xctest.$sSay13JavaScriptKit7JSValueOGACIeggTo_AdCIegxTo_TR (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[314]:0x5719c)
    at JavaScriptKitPackageTests.xctest.$sSay13JavaScriptKit7JSValueOGACIeggTo_AdCIegxTo_TRTA (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1449]:0x9173e)
    at JavaScriptKitPackageTests.xctest.$sSay13JavaScriptKit7JSValueOGACIegxTo_AdCIegiTr_TR (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1480]:0x929a1)
    at JavaScriptKitPackageTests.xctest.$sSay13JavaScriptKit7JSValueOGACIegxTo_AdCIegiTr_TRTA (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1489]:0x93255)
    at JavaScriptKitPackageTests.xctest.$sSay13JavaScriptKit7JSValueOGACIegiTr_AdCIegxTo_TR (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1484]:0x92c96)
    at JavaScriptKitPackageTests.xctest.$sSay13JavaScriptKit7JSValueOGACIegiTr_AdCIegxTo_TRTA.34 (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1493]:0x93456)
    at JavaScriptKitPackageTests.xctest.$s13JavaScriptKit24_call_host_function_implySbSu_SPySo10RawJSValueaGs5Int32Vs6UInt32VtF (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1544]:0x95a63)
    at JavaScriptKitPackageTests.xctest._call_host_function_impl (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[1543]:0x955a8)
    at JavaScriptKitPackageTests.xctest.swjs_call_host_function (wasm://wasm/JavaScriptKitPackageTests.xctest-0f5b6106:wasm-function[5259]:0x17e9a8)
    at SwiftRuntime.callHostFunction (file:///home/katei/ghq/github.com/swiftwasm/JavaScriptKit/.build/plugins/PackageToJS/outputs/PackageTests/runtime.js:414:46)
    at func (file:///home/katei/ghq/github.com/swiftwasm/JavaScriptKit/.build/plugins/PackageToJS/outputs/PackageTests/runtime.js:611:48)

Node.js v22.9.0
error: Error: Test failed with status 1
make: *** [Makefile:10: unittest] Error 1
```

--- a/utils/swift_build_support/swift_build_support/products/wasmswiftsdk.py
+++ b/utils/swift_build_support/swift_build_support/products/wasmswiftsdk.py
@@ -69,9 +69,12 @@ class WasmSwiftSDK(product.Product):
             'CMAKE_Swift_COMPILER_TARGET', swift_host_triple)
         swift_testing.cmake_options.define('CMAKE_SYSROOT', wasi_sysroot)
         swift_resource_dir = os.path.join(dest_dir, 'usr', 'lib', 'swift_static')
+        # For statically linked objects in an archive, we have to use singlethreaded
+        # LLVM codegen unit to prevent runtime metadata sections from being stripped
+        # at link-time.
         swift_testing.cmake_options.define(
             'CMAKE_Swift_FLAGS',
-            f'-sdk {wasi_sysroot} -resource-dir {swift_resource_dir}')
+            f'-sdk {wasi_sysroot} -resource-dir {swift_resource_dir} -Xfrontend -enable-single-module-llvm-emission')
         clang_resource_dir = os.path.join(dest_dir, 'usr', 'lib', 'clang')
         swift_testing.cmake_options.define(
             'CMAKE_CXX_FLAGS', f'-resource-dir {clang_resource_dir}')

